### PR TITLE
Keep the exposed API as putObject

### PR DESCRIPTION
**Pull request by harshavardhana**
exposed API should match our other libraries. Also default SDK tests to run on play.minio.io

### CODE REVIEW

**Comment by krisis** [comment]
@harshavardhana running functional tests on play.minio.io is unacceptably slow. What in the current appproach necessitates a change to play.minio.io server?

``putObjectFromSource`` is named that way to indicate to users that the input is a [conduit](https://github.com/snoyberg/conduit) `Source`. `putObject` doesn't convey that.

**Comment by harshavardhana** [comment]
> @harshavardhana running functional tests on play.minio.io is unacceptably slow. What in the current appproach necessitates a change to play.minio.io server?

The problem is overall experience of forcing users to download Minio, otherwise tests fail it just makes it complicated. Using minio-hs or even writing code shouldn't have a dependency on minio server. If it has such a dependency we should make it automated i.e tests at first bootstrap and download the server - configure it. Then run the tests.

> putObjectFromSource is named that way to indicate to users that the input is a conduit Source. putObject doesn't convey that.

Is this a common norm? my only issue was consistency. If its a common pattern then will change it back.

**Comment by harshavardhana** [comment]
> The problem is overall experience of forcing users to download Minio, otherwise tests fail it just makes it complicated. Using minio-hs or even writing code shouldn't have a dependency on minio server. If it has such a dependency we should make it automated i.e tests at first bootstrap and download the server - configure it. Then run the tests.

Let me explore Travis option to have cached artifacts so that we can avoid download Minio server all the time.

File: src/Network/Minio/Data.hs [hunk 1]
```diff
@@ -27,7 +27,7 @@ data ConnectInfo = ConnectInfo {
   } deriving (Eq, Show)
 
 instance Default ConnectInfo where
-  def = ConnectInfo "localhost" 9000 "minio" "minio123" False "us-east-1"
+  def = ConnectInfo "play.minio.io" 9000 "Q3AM3UQ867SPQQA43P2F" "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG" True "us-east-1"
```
Comment (donatello) [inline src/Network/Minio/Data.hs]: After thinking about this, I think we can go with Minio Play as the default server for tests - it works more simply though it is a bit slow. However, we can add a flag to use a local server for development (I'll take care of this part).

**Comment by donatello** [comment]
> Let me explore Travis option to have cached artifacts so that we can avoid download Minio server all the time.

I think this would be even better if it will work - but we need the option to at some point also download a new minio server (in case of bug fix, etc) overriding the cached artifacts.

**Comment by donatello** [comment]
Also this PR is conflicting because I have split out the tests into two parts. One that uses a live server and others that do not.

**Comment by donatello** [comment]
Closing - changes were accepted with some minor modifications in https://github.com/minio/minio-hs/commit/fa527afa96ba6c70b6a300a6567c8fe3f8adbb6a

**Pull request closed without merging**